test/unit/getTickFormat.spec.js

```diff
@@ -0,0 +1,667 @@
+        expect(ticks[i]).to.be.equal('');
+      } else {
+        expect(ticks[i]).to.be.equal(ticks[i]);
+      }
+    }
+    expect(ticks[ticks.length - 1]).to.be.equal('NA'); // last tick = 'NA' 
+  });
+
+  it('1.3 More than 5 ticks when no NA and has outliers.', function() {
+    ticks = [];
+    logScale = false;
+    data = {
+      'noGrouping': false,
+      'hasNA': false,
+      'min': '',
+      'max': ''
+    };
+
+    opts = {
+      'xDomain' : [0, 1, 2, 3, 4, 5],
+      'emptyMappingVal' : '',
+    };
+
+    data.min = _.min(opts.xDomain);
+    data.max = _.max(opts.xDomain);
+
+    _.each(opts.xDomain, function(v) {
+      ticks.push(util.getTickFormat(v, logScale, data, opts));
+    });
+
+    // console.log(opts.xDomain, ticks);
+
+    expect(ticks).to.have.lengthOf(opts.xDomain.length); // same length
+    expect(ticks[0]).to.be.equal('<=' + opts.xDomain[1]); // first tick = '<=' + value of second tick
+    expect(ticks[ticks.length - 1]).to.be.equal('>' + opts.xDomain[opts.xDomain.length - 2]); // last tick = '>' + value of last second tick
+  });
+
+  it('7. Special case for year: min year > 1500, and length > 7.', function() {
+    ticks = [];
+    logScale = false;
+    data = {
+      'noGrouping': false,
+      'hasNA': false,
+      'min': '',
+      'max': ''
+    };
+
+    opts = {
+      'xDomain' : [1600, 1700, 1800, 1900, 2000, 2100, 2200, 2300, 2400, 2500],
+      'emptyMappingVal' : '',
+    };
+
+    data.min = _.min(opts.xDomain);
+    data.max = _.max(opts.xDomain);
+
+    _.each(opts.xDomain, function(v) {
+      ticks.push(util.getTickFormat(v, logScale, data, opts));
+    });
+
+    // console.log(opts.xDomain, ticks);
+
+    expect(ticks).to.have.lengthOf(opts.xDomain.length); // same length
+    expect(ticks[0]).to.be.equal('<=' + opts.xDomain[1]); // first tick = '<=' + value of second tick
+    expect(ticks[ticks.length - 1]).to.be.equal('>' + opts.xDomain[opts.xDomain.length - 2]); // last tick = '>' + value of last second tick
+  });
+
+  it('7.1 Special case for year: min year > 1500, and length < 7.', function() {
+    ticks = [];
+    logScale = false;
+    data = {
+      'noGrouping': false,
+      'hasNA': false,
+      'min': '',
+      'max': ''
+    };
+
+    opts = {
+      'xDomain' : [1600, 1700, 1800, 1900, 2000],
+      'emptyMappingVal' : '',
+    };
+
+    data.min = _.min(opts.xDomain);
+    data.max = _.max(opts.xDomain);
+
+    _.each(opts.xDomain, function(v) {
+      ticks.push(util.getTickFormat(v, logScale, data, opts));
+    });
+
+    // console.log(opts.xDomain, ticks);
+
+    expect(ticks).to.have.lengthOf(opts.xDomain.length); // same length
+    expect(ticks[0]).to.be.equal('<=' + opts.xDomain[1]); // first tick = '<=' + value of second tick
+    expect(ticks[ticks.length - 1]).to.be.equal('>' + opts.xDomain[opts.xDomain.length - 2]); // last tick = '>' + value of last second tick
+  });
+
+  it('7.2 Special case for year; min year < 1500, and length > 7.', function() {
```

Comment:
I think we mentioned once on setting up the threshold on the year. Could you remind me why we need to do that?

Reply:
I didn't created the threshold but I guess it's used for label of ticks not overlapping.